common/app/views/fragments/social.scala.html

```diff
@@ -1,24 +1,36 @@
-@(sharelinks: Seq[model.ShareLink], position: String = "bottom", iconModifier: String = "", isGallery: Boolean = false)
+@(sharelinks: model.ShareLinkMeta, position: String = "bottom", iconModifier: List[String] = Nil)
 
 <ul class="social social--@position js-social--@position u-unstyled u-cf" data-component="social">
-    @sharelinks.map { item: model.ShareLink =>
-        <li class="social__item social__item--@item.css" data-link-name="@item.css">
-            <a  class="social__action js-social__action--@position social-icon-wrapper"
-                data-link-name="social @position"
-                href="@item.href"
-                target="_blank"
-                title="@item.text">
-                <span class='@if(item.css.matches("facebook|twitter|email")) {inline-icon__fallback button share-modal__item share-modal__item--@item.css} else { u-h }'>@item.userMessage</span>
-                @defining(
-                    if(isGallery){
-                        List("rounded-icon", "social-icon", "social-icon--" + item.css, "social-icon--media")
-                    } else {
-                        List("rounded-icon", "social-icon", "social-icon--" + item.css)
-                    }
-                ) { classNames =>
-                    @fragments.inlineSvg("share-" + item.css, "icon", classNames)
-                }
-            </a>
-        </li>
+    @sharelinks.visible.map { link => @ShareLink(Nil, link) }
+    @if(sharelinks.hidden.isEmpty) { Html("") } else {
+        @SocialMore
+        @sharelinks.hidden.map { link => @ShareLink(List("social--hidden", "social__secondary"), link) }
     }
+    @SaveForLater
+    @CloseTray
 </ul>
+
+@ShareLink(extraClasses: List[String], item: model.ShareLink) = {
+    <li class="social__item social__item--@item.css @extraClasses.mkString(" ")" data-link-name="@item.css">
+        <a  class="social__action js-social__action--@position social-icon-wrapper"
+        data-link-name="social @position"
+        href="@item.href"
+        target="_blank"
+        title="@item.text">
+            <span class='@if(item.css.matches("facebook|twitter|email")) {inline-icon__fallback button share-modal__item share-modal__item--@item.css} else { u-h }'>@item.userMessage</span>
+            @fragments.inlineSvg("share-" + item.css, "icon", List("rounded-icon", "social-icon", "social-icon--" + item.css) ++ iconModifier)
+        </a>
+    </li>
+}
+
+@SocialMore = {
+    <li class="social__secondary social__item social--hidden social__item--more"><span class="social-icon social-icon--more">&bull;&bull;&bull;</span></li>
```

Comment:
This should be button with hidden text as right now it's not tabble by keyboard and doesn't explain what it is via screen reader - therefore the hidden buttons are inaccessible. I would also use an SVG over the `&bull;` as it'd be strange for a screen reader to read them out.